qcodes/instrument_drivers/QuantumDesign/DynaCoolPPMS/private/commandhandler.py

```diff
@@ -0,0 +1,124 @@
+from typing import Any, Tuple, List
+import logging
+from collections import namedtuple
+
+log = logging.getLogger(__name__)
+
+try:
+    import win32com.client
+    import pythoncom
+    from pythoncom import VT_BYREF, VT_R8, VT_I4
+except ImportError as e:
+    message = ('To use the DynaCool Driver, please install win32com.'
+               ' Installation can be done with pip install pypiwin32com')
+    log.exception(message)
+    raise ImportError(message)
+
+
+# Variable types
+_variants = {'VT_R8': win32com.client.VARIANT(VT_BYREF | VT_R8, 0.0),
+             'VT_I4': win32com.client.VARIANT(VT_BYREF | VT_I4, 0)}
+CmdArgs = namedtuple('cmd_and_args', 'cmd args')
+
+
+class CommandHandler:
+    """
+    This is the class that gets called by the server.py
+
+    This class is responsible for making the actual calls into the instrument
+    firmware. The idea is that this class get a SCPI-like string from the
+    server, e.g. 'TEMP?' or 'TEMP 300, 10, 1' and then makes the corresponding
+    MultiVu API call (or returns an error message to the server).
+    """
+
+    def __init__(self, inst_type: str='dynacool') -> None:
+        pythoncom.CoInitialize()
+        client_id = f'QD.MULTIVU.{inst_type.upper()}.1'
+        try:
+            self._mvu = win32com.client.Dispatch(client_id)
+        except pythoncom.com_error:
+            error_mssg = ('Could not connect to Multivu Application. Please '
+                          'make sure that the Multi Application is running.')
+            log.exception(error_mssg)
+            raise ValueError(error_mssg)
+
+        # Hard-code what we know about the MultiVu API
+        self._gets = {'TEMP': CmdArgs(cmd=self._mvu.GetTemperature,
+                                      args=[_variants['VT_R8'],
+                                            _variants['VT_I4']])}
+
+        self._sets = {'TEMP': self._mvu.SetTemperature}
+
+    def preparser(self, cmd_str: str) -> Tuple[CmdArgs, bool]:
```

Comment:
could you rename `pre`/`post`-parsers to smth more explicit like `get_mvu_api_call_thing` and `format_mvu_api_output`? (maybe it's just me, but when I was reading `__call__` method, I had to come back to `preparser` and `postparser` to see what they parse)

Reply:
hashtag wontfix